spy.go

```diff
@@ -1,19 +1,78 @@
 package spy
 
 import (
+	"bufio"
 	"fmt"
 	"io"
+	"os"
 	"os/exec"
+	"os/user"
+	"time"
 
 	"bitbucket.org/creachadair/shell"
+	"github.com/fatih/color"
 )
 
-// ParseCommand accepts a string and returns the command name and args
+// sorry no American English, not today >.<
+type Colour color.Attribute
+
+const (
+	Cyan   Colour = 46
+	Yellow Colour = 43
+	Red    Colour = 101
+	Green  Colour = 42
+	White  Colour = 107
+)
+
+type Option func(*session) error
+
+type session struct {
+	TimestampMode bool
+	ShellPrompt   string
+	Colour        Colour
+	Input         io.Reader
+	Recorder      io.Writer
+	Terminal      io.Writer
+}
+
+func WithTimestamps() Option {
+	return func(s *session) error {
+		s.TimestampMode = true
+		return nil
+	}
+}
+
+func WithUserPrompt(userPrompt string) Option {
+	return func(s *session) error {
+		s.ShellPrompt = userPrompt
+		return nil
+	}
+}
+
+func WithTerminalColour(userColour Colour) Option {
+	return func(s *session) error {
+		switch userColour {
+		case Yellow:
+			s.Colour = Yellow
+		case Red:
+			s.Colour = Red
+		case Green:
+			s.Colour = Green
+		case White:
+			s.Colour = White
+		default:
+			s.Colour = Cyan
+		}
+		return nil
+	}
+}
+
+// ParseCommand accepts a string and returns a command name and a slice of args
 func ParseCommand(cmd string) (string, []string, error) {
 	stringResult, ok := shell.Split(cmd)
 
 	if !ok {
-		return "", []string{}, fmt.Errorf("Cannot parse command: %q", cmd)
+		return "", []string{}, fmt.Errorf("cannot parse command: %q", cmd)
```

Comment:
We can say more, can't we? Something like "unbalanced or unclosed quotes in command line: %q".